### ISSUE TITLE
Streams starting automatically when sltg is up or in background

### ISSUE THREAD

**Issue by faedwen** [streamlink/streamlink-twitch-gui#366]
### Description

more or less what's in the title...
this started to occur when i switched from lstg to sltg. nothing else was changed.

### Reproduction steps

1. start sltg
2. wait till random followed stream goes online
3. stream will start automatically with chosen player. in my case its vlc

### Environment details (operating system, etc.)

w10, latest streamlink, sltg and vlc are the latest versions available

**Comment by bastimeyer** [comment]
This sounds strange... There's actually no way that this can happen unless the selected notification provider is executing a notification click callback. There are no automated stream launch mechanics implemented. Are you sure that you are not clicking any notifications? Are there settings that can cause notifications to be clicked automatically on Windows or can this be done with other programs?


On Win10 the default provider is [SnoreToast](https://github.com/bastimeyer/snoretoast/tree/binaries). As you can see here, the click callback does only get executed if the SnoreToast child process has exited with the success code:
https://github.com/streamlink/streamlink-twitch-gui/blob/v1.0.0/src/app/utils/notification/NotificationProviderSnoreToast.js#L79-L87
https://github.com/bastimeyer/snoretoast/blob/binaries/src/snoretoasts.h#L47-L53

The Growl provider also doesn't execute click callbacks randomly:
https://github.com/streamlink/streamlink-twitch-gui/blob/v1.0.0/src/app/utils/notification/NotificationProviderGrowl.js#L73-L75

When using rich notifications, an event emitted by Chromium is needed for the callback to be executed:
https://github.com/streamlink/streamlink-twitch-gui/blob/v1.0.0/src/app/utils/notification/NotificationProviderRich.js#L33-L39

**Comment by bastimeyer** [comment]
Is it possible that you're using an antivirus software that is interfering with the snoretoast child process?

**Comment by faedwen** [comment]
€dit 1  : sorry for the late answer.

hi there, im very sure that i do not click on notifications because it just happened while i was at work.
and im quite sure that notifications are disabled.
as i returned there were 4 streams opened and running.

im using kaspersky total security 2016.
the wierd thing about this is, that it started to happen, when i migrated from lstg 16 to sltg 1.0.
i think ill try a complete uninstall and reinstall of both (gui and ls). maybe thats gonna help.
and i dont think that u can do much about it. i just wanted to share my experience.

and thx for ur quick reaction
€dit 2 :  and oc for your great and passionate work you put into this project.

**Comment by bastimeyer** [comment]
Can you please run this in the Windows powershell next time your video player opens?
```powershell
gwmi win32_process | % { "$($_.ProcessID) $($_.ParentProcessID) $($_.ProcessName)" }
```
This lists all process IDs and their parent processes and should look like this:
```
7760 1720 streamlink-twitch-gui.exe
4548 7760 streamlink-twitch-gui.exe
2628 7760 streamlink-twitch-gui.exe
260 7760 streamlink-twitch-gui.exe
6716 260 pythonw.exe
6832 6716 mpc-hc64.exe
```
What's important is that the player process and all of its parents are included.

**Comment by faedwen** [comment]
i edited out a bunch of chrome and svchost entries. just for the readabillity.
if u want or need the full list i can edit it in.

0 0 System Idle Process
4 0 System
588 4 smss.exe
820 728 csrss.exe
944 728 wininit.exe
952 928 csrss.exe
1020 944 services.exe
100 944 lsass.exe
1108 928 winlogon.exe
1208 1108 dwm.exe
1592 1020 nvvsvc.exe
1624 1020 igfxCUIService.exe
1776 1592 nvxdsync.exe
2180 1020 spoolsv.exe
2584 1224 dasHost.exe
2768 1020 avp.exe
2788 1020 ETDService.exe
2796 1020 KillerService.exe
2820 1020 PowerlineService.exe
2936 1020 LogiRegistryService.exe
2960 1020 ChargeService.exe
2972 1020 OriginWebHelperService.exe
3040 1020 HeciServer.exe
3052 1020 MSIService.exe
3068 1020 NetworkLicenseServer.exe
816 1020 TBear.Maintenance.exe
2232 1020 SearchIndexer.exe
3316 4 Memory Compression
4092 1020 wmpnetwk.exe
4632 2788 ETDCtrl.exe
4748 1264 sihost.exe
4756 844 WmiPrvSE.exe
4872 1264 taskhostw.exe
4912 1264 OpenHardwareMonitor.exe
4988 1020 PresentationFontCache.exe
5292 844 RuntimeBroker.exe
5568 5500 explorer.exe
5576 5412 igfxEM.exe
5596 5412 igfxHK.exe
6000 5568 ClassicStartMenu.exe
6072 4632 ETDCtrlHelper.exe
6252 844 ShellExperienceHost.exe
6264 2768 avpui.exe
6748 844 SearchUI.exe
6552 844 SkypeHost.exe
8040 844 SettingSyncHost.exe
7500 1776 nvtray.exe
1864 5568 Radio Manager.exe
3984 5568 SCM.exe
1240 5568 rundll32.exe
4340 5568 LCore.exe
8384 5568 RAVCpl64.exe
8732 5568 Steam.exe
8900 5568 NetworkManager.exe
8996 8836 SBCinema.exe
8240 8836 Super-Charger.exe
9068 1020 SteamService.exe
9696 4904 IAStorIcon.exe
1280 1020 IAStorDataMgrSvc.exe
2124 1020 IntelMeFWService.exe
3172 1020 Jhi_service.exe
11840 2448 streamlink-twitch-gui.exe
9364 11840 streamlink-twitch-gui.exe
11904 11840 streamlink-twitch-gui.exe
3544 11840 streamlink-twitch-gui.exe
11720 2028 audiodg.exe
11456 5568 powershell.exe
1300 11456 conhost.exe
5656 8732 steamwebhelper.exe
9260 5656 steamwebhelper.exe
8172 3544 pythonw.exe
10620 8172 vlc.exe
5700 844 WmiPrvSE.exe

**Comment by bastimeyer** [comment]
You can remove all of the unnecessary stuff. I wanted to make sure that the VLC process (and its python parent process) was started by the GUI.

> but now i turned it off

Did you run the powershell command after turning notifications off? You edited your msg, so I don't know.
If they were running and the player started automatically, please set the notification click action to "Do nothing" and keep it running.

**Comment by faedwen** [comment]
i didnt want to do a double post. sorry for causing more trouble.
notifications were on when i ran the command.

now they are back on, with automatic selection, show all but except disabled ones and do nothing

**Comment by bastimeyer** [comment]
Okay, I found the source of the issue.
As I've already guessed, this is caused by the SnoreToast notification provider.

The SnoreToast process incorrectly returns the `0` exit code (success) even if the notification was suppressed by Windows by disabling notifications at all (or the app) in the Windows 10 settings.

I didn't notice this while implementing this, because I didn't test the behavior of disabled notifications when there was a click callback.

Let me fix this and then release `1.0.1` tomorrow...
